client-v2/src/shared/actions/ArticleFragments.ts

```diff
@@ -177,20 +177,20 @@ const getArticleEntitiesFromDrop = async (
   const isURL = isValidURL(resourceIdOrUrl);
   const id = isURL ? getIdFromURL(resourceIdOrUrl) : resourceIdOrUrl;
   const isNonGuLink = isURL && !id;
-  if (isNonGuLink) {
+  const meta = getArticleFragmentMetaFromUrlParams(resourceIdOrUrl);
+  if (isNonGuLink && !meta) {
     const fragment = await createLinkSnap(resourceIdOrUrl);
     return [fragment];
   }
-  if (!id) {
-    return [];
-  }
   try {
-    const meta = getArticleFragmentMetaFromUrlParams(resourceIdOrUrl);
     if (meta) {
       // If we have gu params in the url, create a snap with the meta we extract.
       const fragment = await createLinkSnap(id, meta);
```

Comment:
Could this be renamed to more generic `createSnap`. Assumed link had some meaning here but should be more generic

Reply:
👍